metadata.go

```diff
@@ -192,6 +200,73 @@ func GetImageToMDMap(imageMDs []ImageMetadataInfo) (imageToMDMap map[string][]Im
 	return
 }
 
+// getLocalImages queries the local Docker daemon for list of images.
+func getLocalImages() (imageMap map[ImageIDType][]RepoTagType, e error) {
+
+	// query a list of images from Docker daemon
+	response, e := listImages()
+	if e != nil {
+		return nil, e
+	}
+	blog.Info(string(response))
+	// parse JSON
+	var localImageList []LocalImageStruct
+	if e = json.Unmarshal(response, &localImageList); e != nil {
+		return nil, e
+	}
+
+	// make map from each imageID to all of its aliases (repo+tag)
+	imageMap = make(map[ImageIDType][]RepoTagType)
+	for _, localImage := range localImageList {
+		imageID := ImageIDType(localImage.ID)
+		for _, repoTag := range localImage.RepoTags {
```

Comment:
Sometimes there are images with no repo:tag, in which case "docker images" prints "<none>" for the repo and the tag, like this:

REPOSITORY                                  TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
<none>                                      <none>              a5573d9ea98b        5 days ago          792.6 MB

Collector should probably ignore these images.

Also, when Collector pulls images from a registry, it may grab only from a subset of the repositories as specified in the ReposToProcess map. If the map isn't empty, could you use it to filter the set of images your code handles?